File: livingrimoire start here/LivinGrimoire java/src/LivinGrimoire/Mutatable.java
package LivinGrimoire;

public abstract class Mutatable {
    // one part of an algorithm, it is a basic simple action or sub goal
    public Boolean algKillSwitch = false;
    public abstract String action(String ear, String skin, String eye);
    public abstract Boolean completed();
    public  String myName(){
        // Returns the class name
        return this.getClass().getSimpleName();
    }

}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/livinGrimoire/Mutatable.kt
package livinGrimoire

open class Mutatable {
    // one part of an algorithm, it is a basic simple action or sub goal
    var algKillSwitch = false
    open fun action(ear: String, skin: String, eye: String): String {return  ""}
    open fun completed(): Boolean {return false}
    fun myName(): String {
        // Returns the class name
        return this.javaClass.simpleName
    }
}